docs/vspec_arch.md

```diff
@@ -1,4 +1,4 @@
-# vspec2x Design Decisions, Architecture and Limitations
+# vspec Design Decisions, Architecture and Limitations
```

Comment:
I am thinking if we possibly should add a few lines in this file saying something about our lazy-loading/generator concept. 

_(In this PR we remove the  `example_generator` test which is fine for me, but maybe it would be beneficial later in a different PR to add a tutorial/example on how you could add your own generator/exporter, i.e. how you could introduce something similar to the example in the obsolete  `example_generator`)_

Reply:
Done